week03_model_free/display_server_preparation.sh

```diff
@@ -0,0 +1,47 @@
+echo "apt-get updating..."
+apt-get -qq update > /dev/null
+echo "x11vnc installing if it is not exists..."
+echo ""
+apt-get -q install -y x11vnc
+echo ""
+echo "x11vnc is installed"
+checkXvfbProcess=1 # 0 == True, 1 == False
+if [ "$checkXvfbProcess" -eq 0 ];
```

Comment:
`checkXvfbProcess` only exists during the execution of this script, and it is set to `1`, so this condition is _never_ satisfied.

Reply:
Yes, it is a my point. If an user is brave enough to open the script he can change "1" on "0" for checking that xvfb, not other display provider, is used. Also xdpyinfo checking can fail some time later but there is one more way to check. It is rare cases, of course, but another aim of this _never_ satisfied condition and checking to educate the brave user to do such things later. I have not such examples, and spend a lot of time to implement script. (And only at the end of script inplementation I found "xdpyinfo" when searched a solution of another problem about displays.) Finally, it is not a script for high-loaded systems, thus I suppose we can left another checking way.